pages/api/users/[userId]/index.js

```diff
@@ -0,0 +1,31 @@
+import { Record, String } from 'runtypes'
+import { User } from '../../../../models/user'
+import connectDB from '../../../../middleware/mongodb'
+import handleError from '../../../../utils/handleError'
+
+
+const getUserByIdRunType = Record({
+    query: Record({
+        userId: String
+    })
+})
+
+const handler = async (req, res) => {
+    if (req.method === 'GET') {
+        try {
+            const validatedRequest = getUserByIdRunType.check(req)
+            const { userId } = validatedRequest.query
+
+            const user = await User.findOne({ _id: userId })
+
```

Comment:
What happens if no user exists? Does it throw an error or return null? if it returns null, you need to throw an error yourself